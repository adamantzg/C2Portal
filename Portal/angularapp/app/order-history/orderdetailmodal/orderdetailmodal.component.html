<div class="modal-header">
    <h3 class="modal-title" id="modal-title">{{order.order_no}} details</h3>
  </div>
  <div class="modal-body" id="modal-body">
    <div class="alert alert-danger" *ngIf="errorMessage">{{errorMessage}}</div>
    <table class="table table-striped">
        <thead>
            <tr>
                <th>Product</th>
                <th class="text-right">Qty</th>
                <th class="text-right">Value</th>
            </tr>        
        </thead>
        <tbody>
            <tr *ngFor="let d of order.details">
                <td>{{d.product}}</td>
                <td class="text-right">{{d.order_qty}}</td>
                <td class="text-right">{{d.val | number:numberFormat}}</td>
            </tr>
        </tbody>
        <tfoot>
            <tr>
                <td colspan="2"></td>
                <td class="text-right" style="font-weight: bold">{{total | number:numberFormat}}</td>
            </tr>            
        </tfoot>
    </table>
  </div>
  
  <div class="modal-footer">
      <button class="btn btn-primary" type="button" (click)="close()" >Close</button>    
  </div>
  


