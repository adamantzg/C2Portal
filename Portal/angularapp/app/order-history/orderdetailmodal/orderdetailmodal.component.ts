import { Component, OnInit } from '@angular/core';
import { Order } from '../../domainclasses';
import { BsModalRef } from 'ngx-bootstrap';

@Component({
  selector: 'app-orderdetailmodal',
  templateUrl: './orderdetailmodal.component.html',
  styleUrls: ['./orderdetailmodal.component.css']
})
export class OrderdetailmodalComponent implements OnInit {

  constructor(private bsModalRef: BsModalRef) { }

  order: Order = new Order();
  total = 0;
  errorMessage = '';
  numberFormat = '1.2-2';

  ngOnInit() {
  }

  close() {
    this.bsModalRef.hide();
  }

}
