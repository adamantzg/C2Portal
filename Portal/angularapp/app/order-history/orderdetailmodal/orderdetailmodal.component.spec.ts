import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import { OrderdetailmodalComponent } from './orderdetailmodal.component';
import { BsModalRef, ModalModule } from 'ngx-bootstrap';

describe('OrderdetailmodalComponent', () => {
  let component: OrderdetailmodalComponent;
  let fixture: ComponentFixture<OrderdetailmodalComponent>;

  setupTestBed({
      declarations: [ OrderdetailmodalComponent ],
      providers: [BsModalRef],
      imports: [ModalModule]
    });

  beforeEach(() => {
    fixture = TestBed.createComponent(OrderdetailmodalComponent);
    component = fixture.componentInstance;

  });

  test('should create', () => {
    expect(component).toBeTruthy();
    expect(component).toMatchSnapshot();
  });
});
