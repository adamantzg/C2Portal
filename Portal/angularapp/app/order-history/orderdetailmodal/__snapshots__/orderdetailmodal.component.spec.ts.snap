// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderdetailmodalComponent should create 1`] = `
OrderdetailmodalComponent {
  "bsModalRef": BsModalRef {
    "hide": [Function],
  },
  "errorMessage": "",
  "numberFormat": "1.2-2",
  "order": Order {},
  "total": 0,
}
`;
